### PR TITLE
Folder Structure for GlobalConfiguration CurOffTimes and DbContext

### PR DESCRIPTION
**Pull request by mihir1678**

### CODE REVIEW

File: CIR.Data/Data/GlobalConfiguration/GlobalConfigurationCutOffTimesRepository.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        ///</summary>
+        ///<param name="globalConfigurationCutOffTimeModel"></param>
+		/// <returns>Success status if its valid else failure</returns>
+        public async Task<IActionResult> CreateOrUpdateGlobalConfigurationCutOffTime(GlobalConfigurationCutOffTimeModel globalConfigurationCutOffTimeModel)
+        {
+            try
+            {
+                if (commonRepository.IsStringNullorEmpty(globalConfigurationCutOffTimeModel.CutOffTime) || globalConfigurationCutOffTimeModel.CountryId == 0)
+                {
+                    return new JsonResult(new CustomResponse<string>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.BadRequest, Result = false, Message = HttpStatusCodesAndMessages.HttpStatus.BadRequest.GetDescriptionAttribute(), Data = SystemMessages.msgEnterValidData });
```
Comment (GohilAmit) [inline CIR.Data/Data/GlobalConfiguration/GlobalConfigurationCutOffTimesRepository.cs:91]: need to change return response message

File: CIR.Data/Data/GlobalConfiguration/GlobalConfigurationCutOffTimesRepository.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                        parameters.Add("@CutOffDay", globalConfigurationCutOffTimeModel.CutOffDay);
+
+                        result = (int)await Task.FromResult(connection.ExecuteScalar("spCreateOrUpdateGlobalConfigurationCutOffTimes", parameters, commandType: CommandType.StoredProcedure));
+                    }
+                }
+                if (result != 0)
+                {
+                    if (globalConfigurationCutOffTimeModel.Id > 0)
+                    {
+                        return new JsonResult(new CustomResponse<string>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.Success, Result = true, Message = HttpStatusCodesAndMessages.HttpStatus.Success.GetDescriptionAttribute(), Data = string.Format(SystemMessages.msgDataUpdatedSuccessfully, "GlobalConfiguration CutOffTimes") });
```
Comment (GohilAmit) [inline CIR.Data/Data/GlobalConfiguration/GlobalConfigurationCutOffTimesRepository.cs:111]: need to change return response message

File: CIR.Data/Data/GlobalConfiguration/GlobalConfigurationCutOffTimesRepository.cs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                    }
+                    else
+                    {
+                        return new JsonResult(new CustomResponse<string>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.Success, Result = true, Message = HttpStatusCodesAndMessages.HttpStatus.Success.GetDescriptionAttribute(), Data = string.Format(SystemMessages.msgDataSavedSuccessfully, "GlobalConfiguration CutOffTimes") });
```
Comment (GohilAmit) [inline CIR.Data/Data/GlobalConfiguration/GlobalConfigurationCutOffTimesRepository.cs:115]: need to change return response message

File: CIR.Data/Data/GlobalConfiguration/GlobalConfigurationCutOffTimesRepository.cs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                    }
+                }
+                return new JsonResult(new CustomResponse<string>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.UnprocessableEntity, Result = false, Message = HttpStatusCodesAndMessages.HttpStatus.UnprocessableEntity.GetDescriptionAttribute(), Data = SystemMessages.msgBadRequest });
```
Comment (GohilAmit) [inline CIR.Data/Data/GlobalConfiguration/GlobalConfigurationCutOffTimesRepository.cs:118]: need to change return response message

File: CIR/Controllers/GlobalConfiguration/GlobalConfigurationCutOffTimesController.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        [HttpPost("[action]")]
+        public async Task<IActionResult> Create([FromBody] GlobalConfigurationCutOffTimeModel globalConfigurationCutOffTimeModel) 
+        {
+            if(ModelState.IsValid)
+            {
+                try
+                {
+                    return await iGlobalConfigurationCutOffTimesService.CreateOrUpdateGlobalConfigurationCutOffTime(globalConfigurationCutOffTimeModel);
+                }
+                catch (Exception ex)
```
Comment (GohilAmit) [inline CIR/Controllers/GlobalConfiguration/GlobalConfigurationCutOffTimesController.cs:63]: need to change catch return response

File: CIR/Controllers/GlobalConfiguration/GlobalConfigurationCutOffTimesController.cs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                {
+                    return new JsonResult(new CustomResponse<Exception>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.InternalServerError, Result = false, Message = HttpStatusCodesAndMessages.HttpStatus.InternalServerError.GetDescriptionAttribute(), Data = ex });
+                }
+            }
+            return new JsonResult(new CustomResponse<string>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.BadRequest, Result = false, Message = HttpStatusCodesAndMessages.HttpStatus.BadRequest.GetDescriptionAttribute(), Data = SystemMessages.msgBadRequest });
```
Comment (GohilAmit) [inline CIR/Controllers/GlobalConfiguration/GlobalConfigurationCutOffTimesController.cs:68]: need to change return response

File: CIR/Controllers/GlobalConfiguration/GlobalConfigurationCutOffTimesController.cs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        {
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    return await iGlobalConfigurationCutOffTimesService.CreateOrUpdateGlobalConfigurationCutOffTime(globalConfigurationCutOffTimeModel);
+                }
+                catch (Exception ex)
+                {
+                    return new JsonResult(new CustomResponse<Exception>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.InternalServerError, Result = false, Message = HttpStatusCodesAndMessages.HttpStatus.InternalServerError.GetDescriptionAttribute(), Data = ex });
```
Comment (GohilAmit) [inline CIR/Controllers/GlobalConfiguration/GlobalConfigurationCutOffTimesController.cs:87]: need to change catch return response

File: CIR/Controllers/GlobalConfiguration/GlobalConfigurationCutOffTimesController.cs [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+                }
+            }
+            return new JsonResult(new CustomResponse<string>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.BadRequest, Result = false, Message = HttpStatusCodesAndMessages.HttpStatus.BadRequest.GetDescriptionAttribute(), Data = SystemMessages.msgBadRequest });
```
Comment (GohilAmit) [inline CIR/Controllers/GlobalConfiguration/GlobalConfigurationCutOffTimesController.cs:90]: need to change return response

File: CIR/Controllers/GlobalConfiguration/GlobalConfigurationCutOffTimesController.cs [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+        [HttpGet("{countryId}")]
+        public async Task<IActionResult> GetGlobalConfigurationCutOffTimeByCountryWise(int countryId)
+        {
+            try
+            {
+                return await iGlobalConfigurationCutOffTimesService.GetGlobalConfigurationCutOffTimeByCountryWise(countryId);
+            }
+            catch (Exception ex)
+            {
+                return new JsonResult(new CustomResponse<Exception>() { StatusCode = (int)HttpStatusCodesAndMessages.HttpStatus.InternalServerError, Result = false, Message = HttpStatusCodesAndMessages.HttpStatus.InternalServerError.GetDescriptionAttribute(), Data = ex });
```
Comment (GohilAmit) [inline CIR/Controllers/GlobalConfiguration/GlobalConfigurationCutOffTimesController.cs:45]: need to change catch return response

**Review by GohilAmit — APPROVED** [review]

**Review by mittalvyas1 — APPROVED** [review]

**Pull request merged**